src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -0,0 +1,116 @@
+            this.AsynchronousPatternEnabled = asynchronousPatternEnabled;
+        }
+
+        internal DurableHttpRequest(JObject jObject)
+        {
+            this.Method = jObject["Method"].ToObject<HttpMethod>();
+            this.Uri = jObject["Uri"].ToObject<Uri>();
+
+            Dictionary<string, StringValues> headerDictStringValues = new Dictionary<string, StringValues>();
+            Dictionary<string, IEnumerable<string>> headersDictEnumerable = jObject["Headers"].ToObject<Dictionary<string, IEnumerable<string>>>();
+            foreach (var header in headersDictEnumerable)
+            {
+                string key = header.Key;
+                string[] headerValues = header.Value.ToArray<string>();
+                StringValues values = new StringValues(headerValues);
+                headerDictStringValues.Add(key, values);
+            }
+
+            this.Headers = headerDictStringValues;
+
+            this.Content = jObject["Content"].Value<string>();
+
+            JsonSerializerSettings serializer = new JsonSerializerSettings();
+            serializer.TypeNameHandling = TypeNameHandling.Auto;
+            string tokenSource = JsonConvert.SerializeObject(jObject["TokenSource"], serializer);
+
+            this.TokenSource = JsonConvert.DeserializeObject<ITokenSource>(tokenSource, serializer);
+            this.AsynchronousPatternEnabled = jObject["AsynchronousPatternEnabled"].Value<bool>();
+        }
+
+        /// <summary>
+        /// HttpMethod used in the HTTP request made by the Durable Function.
+        /// </summary>
+        public HttpMethod Method { get; }
+
+        /// <summary>
+        /// Uri used in the HTTP request made by the Durable Function.
+        /// </summary>
+        public Uri Uri { get; }
+
+        /// <summary>
+        /// Headers passed with the HTTP request made by the Durable Function.
+        /// </summary>
+        public IDictionary<string, StringValues> Headers { get; }
+
+        /// <summary>
+        /// Content passed with the HTTP request made by the Durable Function.
+        /// </summary>
+        public string Content { get; }
+
+        /// <summary>
+        /// Information needed to get a token for a specified service.
+        /// </summary>
+        [JsonProperty(TypeNameHandling = TypeNameHandling.Auto)]
+        public ITokenSource TokenSource { get; }
+
+        /// <summary>
+        /// Specifies whether the Durable HTTP APIs should automatically
+        /// handle the asynchronous HTTP pattern.
+        /// </summary>
+        public bool AsynchronousPatternEnabled { get; } = true;
+
+        private static Dictionary<string, StringValues> CreateDictionaryCopy(IDictionary<string, StringValues> headers)
+        {
+            Dictionary<string, StringValues> newDictionary = new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase);
+            foreach (KeyValuePair<string, StringValues> pair in headers)
```

Comment:
I think you need a check for `null` before you try to enumerate `headers`. Otherwise you'll get a `NullReferenceException` here.